src/MSBuild.Bootstrap/MSBuild.Bootstrap.csproj

```diff
@@ -0,0 +1,40 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\MSBuild\MSBuild.csproj" />
```

Comment:
I don't think we can rely on publish for this since all our unit tests need it. We're essentially doing a publish during build. That plus we aren't expressing all of the run-time references in MSBuild.csproj, and I'm not sure we should. So this is essentially our publish of MSBuild.csproj that happens during the build.

Reply:
It could probably end up being part of MSBuild.csproj, but I think I prefer it separate.  It keeps the MSBuild project simpler, and the bootstrap project doesn't need to be included in MSBuild.Dev.sln.

I think it would make sense to have the full framework bootstrapping also be in a separate project, but since it's working as is I didn't change it.

Reply:
@AndyGerlicher 
From what I've seen, tests run from the output of each unit test assembly, not the bootstrapped directory.

Ideally, it would be nice to just publish MSBuild.csproj. More realistically, if this requires a couples of hacks, I'd rather have the hacks in a separate csproj, like it is now. What I don't like here is that net46 follows a different bootstrap pattern (not published based) than netcoreapp (published based).